src/Features/Core/Portable/Shared/Utilities/LogicalStringComparer.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Shared.Utilities
+{
+    internal class LogicalStringComparer : IComparer<string>
+    {
+        [DllImport("shlwapi.dll", CharSet = CharSet.Unicode)]
```

Comment:
There's a couple of issues here; this a portable dll, this is going to throw on non-Windows desktop platforms. We should move this into a non-portable layer, so that we can have a fall back on non-Windows platforms.

Secondly, for Windows platforms you should be p/invoking to [APIs sets](https://msdn.microsoft.com/en-us/library/windows/desktop/hh802935(v=vs.85).aspx) so that you work portable across the Windows platforms that support this API.

Thirdly, this API isn't available in AppContainer apps (UWP), so you'll need to specifically handle that as well.